src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -974,17 +980,54 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
         auto uri{ fmt::format(L"{}terminals?cols={}&rows={}&version=2019-01-01&shell={}", _cloudShellUri, _initialCols, _initialRows, shellType) };
 
         WWH::HttpStringContent content{
-            L"",
+            L"{}",
             WSS::UnicodeEncoding::Utf8,
             // LOAD-BEARING. the API returns "'content-type' should be 'application/json' or 'multipart/form-data'"
             L"application/json"
         };
 
-        const auto terminalResponse = _SendRequestReturningJson(uri, content);
+        const auto terminalResponse = _SendRequestReturningJson(uri, content, WWH::HttpMethod::Post(), Windows::Foundation::Uri(_cloudShellUri));
         _terminalID = terminalResponse.GetNamedString(L"id");
 
+        auto socketUri = terminalResponse.GetNamedString(L"socketUri");
+
+        // we have to do some post-handling to get the proper socket endpoint
+        // the logic here is based on the way the cloud shell team itself does it
+        winrt::hstring finalSocketUri;
+        std::wstring wCloudShellUri{ _cloudShellUri };
+
+        if (wCloudShellUri.find(L"servicebus") == std::wstring::npos)
+        {
+            // wCloudShellUri does not contain the word "servicebus", we can just use it to make the final URI
+
+            // remove the "https://" from the cloud shell URI
+            std::wstring https{ L"https://" };
+            wCloudShellUri.replace(wCloudShellUri.find(https), https.length(), L"");

```

Comment:
In C++ code, valid usage of `string` and `wstring` is actually pretty rare. In this instance it's the same thing: This allocates 18 bytes of heap, copies the string constant there to make it mutable, and then it throws it immediately away after using it as a search needle.

Personally, during coding I always rely heavily on the language documentation. In case of C++ I pretty much always got cppreference open. Here it would be: https://en.cppreference.com/w/cpp/string/basic_string
That shows, that we can use the `erase()` function instead of `replace()`.

This would be fine IMO:
```cpp
wCloudShellUri.replace(wCloudShellUri.find(L"https://"), 8);
```
because while 8 is a magic number it's correlation with the find() string length is pretty obvious. Alternatively I'd make the https prefix a `static constexpr std::wstring_view` and use its `.size()`.

But we don't need to copy the URI into `wCloudShellUri` at all anyways. We _know_ that the `https://` will only ever be at the start of the string, so we can just trim off the first 5 characters here. Something like this:
```cpp
const auto uriWithoutProtocol = std::wstring_view{ _cloudShellUri }.substr(5);
finalSocketUri = fmt::format(FMT_COMPILE(L"wss{}terminals/{}"), uriWithoutProtocol, _terminalID);
```